astropy/tests/helper.py

```diff
@@ -409,11 +410,18 @@ def generic_recursive_equality_test(a, b, class_history):
     Check if the attributes of a and b are equal. Then,
     check if the attributes of the attributes are equal.
     """
-    dict_a = a.__getstate__() if hasattr(a, '__getstate__') else a.__dict__
+    if hasattr(a, '__getstate__'):
+        if not PYTHON_LT_3_10:
```

Comment:
Original comment disappeared now, but I still don't follow?
> From the comment "This check (and thus the else case will no longer be needed when Python 3.10...". Was the comment wrong?

When 3.10 is _dropped_, i.e. `PYTHON_LT_3_11` always becomes `False`. In 3.10 to my knowledge there also is no case of the `_PyObject_GetState(PyObject *obj)` (with argument) form, at least not the default implementation for `class`.

Reply:
Ohhhh ok. I'll change it.